azure-kusto-data/azure/kusto/data/aio/streaming_response.py

```diff
@@ -0,0 +1,253 @@
+            self.finished_primary_results = True
+
+        return parsed_frame
+
+    async def parse_frame(self, frame_type):
+        if frame_type == FrameType.DataSetHeader:
+            return await self.extract_props(frame_type, ("IsProgressive", JsonTokenType.BOOLEAN), ("Version", JsonTokenType.STRING))
+        elif frame_type == FrameType.TableHeader:
+            return await self.extract_props(
+                frame_type,
+                ("TableId", JsonTokenType.NUMBER),
+                ("TableKind", JsonTokenType.STRING),
+                ("TableName", JsonTokenType.STRING),
+                ("Columns", JsonTokenType.START_ARRAY),
+            )
+        elif frame_type == FrameType.TableFragment:
+            return await self.extract_props(
+                frame_type, ("TableFragmentType", JsonTokenType.STRING), ("TableId", JsonTokenType.NUMBER), ("Rows", JsonTokenType.START_ARRAY)
+            )
+        elif frame_type == FrameType.TableCompletion:
+            return await self.extract_props(frame_type, ("TableId", JsonTokenType.NUMBER), ("RowCount", JsonTokenType.NUMBER))
+        elif frame_type == FrameType.TableProgress:
+            return await self.extract_props(frame_type, ("TableId", JsonTokenType.NUMBER), ("TableProgress", JsonTokenType.NUMBER))
+        elif frame_type == FrameType.DataTable:
+            props = await self.extract_props(
+                frame_type,
+                ("TableId", JsonTokenType.NUMBER),
+                ("TableKind", JsonTokenType.STRING),
+                ("TableName", JsonTokenType.STRING),
+                ("Columns", JsonTokenType.START_ARRAY),
+            )
+            await self.reader.skip_until_property_name("Rows")
+            props["Rows"] = self.row_iterator()
+            if props["TableKind"] != WellKnownDataSet.PrimaryResult.value:
+                props["Rows"] = [r async for r in props["Rows"]]
+            return props
+        elif frame_type == FrameType.DataSetCompletion:
+            res = await self.extract_props(frame_type, ("HasErrors", JsonTokenType.BOOLEAN), ("Cancelled", JsonTokenType.BOOLEAN))
+            token = await self.reader.skip_until_property_name_or_end_object("OneApiErrors")
+            if token.token_type != JsonTokenType.END_MAP:
+                res["OneApiErrors"] = self.parse_array(skip_start=False)
+            return res
+
+    async def row_iterator(self):
+        await self.reader.read_token_of_type(JsonTokenType.START_ARRAY)
+        while True:
+            token = await self.reader.read_token_of_type(JsonTokenType.START_ARRAY, JsonTokenType.END_ARRAY, JsonTokenType.START_MAP)
+            if token.token_type == JsonTokenType.START_MAP:
+                raise KustoServiceError(
+                    "Received error in data: " + str(self.parse_object(skip_start=True))
+                )  # Todo - replace this with a better error once managed streaming ingest is merged
```

Comment:
I'm doing a managed streaming ingest branch, and in it I created more specific errors so it can parse them for its functions, so I added this comment to not forget adding them here

This  is probably premature since that branch's PR isn't up yet